src/luasocket.c

```diff
@@ -51,6 +54,9 @@ static const luaL_Reg mod[] = {
     {"tcp", tcp_open},
     {"udp", udp_open},
     {"select", select_open},
+#ifdef LUASOCKET_NETLINK
+    {"netlink", netlink_open},
+#endif
```

Comment:
It's because the netlink socket is only slightly different than a udp socket, so it seemed fair to not separate them.

Reply:
Even if being similar or different were a good reason to include in the core part of LuaSocket (which is not!) then UNIX domain socket would have to be included as well because it is much more similar to "inet" sockets than Netlink. So you either adopt the same approach for 'socket.unix' or you change the way you added your Netlink support.